client/packages/programs/src/JsonForms/components/Search/SearchWithUserSource.tsx

```diff
@@ -18,140 +16,157 @@ import {
-                  label={t('label.edit')}
-                  icon={<EditIcon style={{ width: 16 }} />}
-                  onClick={() => {
-                    setEditMode(true);
-                  }}
-                  color="primary"
-                  height="20px"
-                  width="20px"
+    <Box>
+      <Typography
+        variant="subtitle1"
+        width={'100%'}
+        textAlign="left"
+        marginBottom={1}
+        paddingBottom={1}
+        paddingTop={3}
+      >
+        <strong>{label}</strong>
+      </Typography>
+      <JsonFormsDispatch
+        schema={schema}
+        uischema={
+          {
+            type: 'VerticalLayout',
+            elements: options.elements,
+          } as UISchemaElement
+        }
+        path={path}
+        renderers={renderers}
+        enabled={!isPatientSelected}
+      />
+      {(isPatientSelected || results.length > 0) && (
+        <DetailInputWithLabelRow
+          sx={DefaultFormRowSx}
+          label=""
+          labelWidthPercentage={FORM_LABEL_WIDTH}
+          inputAlignment={'start'}
+          Input={
+            !isPatientSelected ? (
+              <Box>
+                <Typography variant="body2" mt={1} mb={1}>
+                  <em>{t('control.search.matching-patients')}</em>
+                </Typography>
+                <Select
+                  options={results.map(res => ({
+                    label: getOptionLabel(res) ?? '',
+                    value: res.id,
+                  }))}
+                  onChange={e => handlePatientSelect(e.target.value)}
+                  fullWidth
                 />
-              </>
+              </Box>
             ) : (
-              <Typography color="error">{error}</Typography>
-            )}
-          </Box>
-        )
-      }
-    />
+              <Box
+                display="flex"
+                alignItems="center"
+                justifyContent="space-between"
+                flexBasis="100%"
+                sx={{ width: FORM_INPUT_COLUMN_WIDTH }}
+              >
+                {!error ? (
+                  <Button
+                    onClick={() => {
+                      handleChange(path, {});
+                      resetResults();
+                    }}
+                    variant="outlined"
+                    size="small"
+                    sx={{ mt: 1 }}
+                  >
+                    {t('control.search.reset-button')}
+                  </Button>
+                ) : (
+                  <Typography color="error">{error}</Typography>
+                )}
+              </Box>
+            )
+          }
+        />
+      )}
+    </Box>
   );
 };
+
+// Most search fields will be matched using partial string matching, using
+// "like". However, this is not available/reasonable for some fields, so their
+// match type is referenced here
+const queryMatchTypes = {
+  // Add more as required
+  gender: 'equalTo',
+  dateOfBirth: 'equalTo',
+};
```

Comment:
Have only checked the fields used in this example -- firstName, lastName, code, gender, dateOfBirth. If we want to search by others, we might need to add a match type in here.